src/service/userActivity.service.ts

```diff
@@ -59,41 +56,24 @@ export const getActivityList = async ({
   }
 };
 
-export const attendActivity = async ({
-  userId,
+export const getActivityDetail = async ({
   activityId,
-  requestBody,
-}: AttendActivityParams) => {
-  const user = await User.findById(userId);
-  if (!user) {
-    throw new CoreError('User not found.');
-  }
-
-  if (!user.isEmailValidate) {
-    throw new CoreError('User email not validated.');
-  }
-
-  const activity = await Activity.findById(activityId).populate('tickets');
-
-  if (!activity) {
-    throw new CoreError('Activity not found.');
-  }
-
-  // if (activity.tickets?.some((i) => i.userId.equals(userId))) {
-  //   throw new CoreError('The user already attended the activity.');
-  // }
-
+}: GetActivityDetailCredential) => {
   try {
-    await Ticket.create({
-      userId,
-      activityId,
-      serialNumber: faker.string.uuid(),
-      ticketStatus: TicketStatusEnum.VALID,
-      ...requestBody,
+    const activity = await Activity.find({
+      _id: activityId,
     });
-    return { message: 'Attend activity success.' };
+    const tickets = await Ticket.find({
+      activityId,
+    }).select('-activityId');
+    const data = {
+      activity,
+      tickets,
+    };
```

Comment:
因為 find 回傳的是陣列，要用 activity[0].tickets 才能取 tickets 的資料，單筆活動資料的話，用 findOne 會回傳活動 object，就可以直接取得 tickets 的值
```suggestion
    const activity = await Activity.findOne({
      _id: activityId,
    }).populate('tickets');
    const data = {
      activity,
      tickets: activity?.tickets,
    };
```